File: Python_TRR/13.py
# Viết bằng Python
# Bài 13: Tìm ước chung lớn nhất (GCD) và bội chung nhỏ nhất (LCM).

from math import gcd # import hàm gcd từ thư viện math

def lcm(a, b):
    return a * b // gcd(a, b) # LCM = a * b / GCD của a và b

print("GCD:", gcd(12, 18))  # GCD của 12 và 18
print("LCM:", lcm(12, 18))  # LCM của 12 và 18

File: C_TRR/13.c
// Viết bằng C
// Bài 13: Tìm ước chung lớn nhất (GCD) và bội chung nhỏ nhất (LCM).

#include <stdio.h>

int gcd(int a, int b) // Hàm tìm GCD của 2 số a, b (a > b)
{
    return b == 0 ? a : gcd(b, a % b);  // Đệ quy
}

int lcm(int a, int b)  // Hàm tìm LCM của 2 số a, b (a > b)
{
    return a * b / gcd(a, b);  // Tính LCM từ GCD
}

int main() 
{
    printf("GCD: %d\n", gcd(12, 18)); // GCD: 6 
    printf("LCM: %d\n", lcm(12, 18)); // LCM: 36
    return 0;
}